code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,369 @@
+
+/obj/item/medigunpack/New()
+	..()
+	medigun = new(src)
+
+/obj/item/medigunpack/Destroy()
+	if(medigun)
+		medigun.medigunpack = null
+		qdel(medigun)
+	medigun = null
+	..()
+
+/obj/item/medigunpack/MouseDrop(atom/over_object)
+	if(ishuman(usr))
+		var/mob/living/carbon/human/H = usr
+		if (!istype(over_object, /obj/screen/inventory))
+			return ..()
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+
+		if(medigun)
+			if(istype(medigun.loc,/mob))
+				var/mob/M = medigun.loc
+				M.u_equip(medigun)
+			medigun.forceMove(src)
+			update_icon()
+
+		if ((src == H.get_item_by_slot(slot_back)) && !H.incapacitated())
+			var/obj/screen/inventory/OI = over_object
+
+			if(OI.hand_index)
+				H.u_equip(src, 1)
+				H.put_in_hand(OI.hand_index, src)
+				H.update_inv_wear_suit()
+				add_fingerprint(H)
+
+/obj/item/medigunpack/attack_hand(mob/user)
+	if (medigun && (src.loc == user))
+		add_fingerprint(user)
+		user.put_in_hands(medigun)
+		update_icon()
+	else
+		..()
+
+/obj/item/medigunpack/attackby(var/obj/item/weapon/W, var/mob/user)
+	if(W == medigun)
+		user.drop_item(W,get_turf(src))
+		W.forceMove(W)
+		update_icon()
+		return 1
+	return ..()
+
+/obj/item/medigunpack/emag_act(mob/user as mob)
+	medigun.emag_act(user)
+
+/obj/item/medigunpack/dropped(mob/user)
+	if(medigun)
+		if(istype(medigun.loc,/mob))
+			var/mob/M = medigun.loc
+			M.u_equip(medigun)
+		medigun.forceMove(src)
+		update_icon()
+	else
+		medigun = new(src)
+		medigun.emagged = emagged
+
+/obj/item/medigunpack/stripped(mob/wearer,mob/stripper)
+	dropped(wearer)
+
+/obj/item/medigunpack/update_icon()
+	if(medigun && (medigun.loc == src))
+		icon_state = "medigunfull"
+	else
+		icon_state = "medigunpack"
+	if(emagged)
+		icon_state += "red"
+		item_state = "medigunred"
+
+/obj/item/uberdevice
+	name = "\improper Uber Device"
+	desc = "This small device has a gauge with Uber written on it."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "uber_device"
+	w_class = W_CLASS_SMALL
+	force = 2
+	throwforce = 5
+
+/obj/item/medigun
+	name = "\improper Medi-Gun"
+	desc = "Now aim at people."//I'm tired right now, someone pls find a snarky, possibly TF2-referencing, description to replace that
+	icon = 'icons/obj/gun_experimental.dmi'
+	icon_state = "medigun"
+	item_state = "medigun"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	force = 5
+	throwforce = 5
+	throw_speed = 1
+	throw_range = 1
+	flags = FPRINT
+	w_class = W_CLASS_HUGE
+	var/fire_sound = 'sound/weapons/medigun_heal.ogg'
+	var/empty_sound = 'sound/weapons/medigun_no_target.ogg'
+	var/obj/item/medigunpack/medigunpack = null
+	var/mob/living/healtarget = null
+	var/mob/living/wielder = null
+	var/ubercharge = 0
+	var/emagged = 0
+
+/obj/item/medigun/examine(mob/user)
+	..()
+	to_chat(user,"<span class='info'>Charge = [ubercharge]%</span>")
+
+/obj/item/medigun/New()
+	..()
+	medigunpack = loc
```

Comment:
This is horrendously unsafe. Make it get passed as an argument to New() instead.